src/Microsoft.VisualStudio.Services.Agent/Blob/DedupManifestArtifactClientFactory.cs

```diff
@@ -221,5 +238,36 @@ public static IAppTraceSource CreateArtifactsTracer(bool verbose, Action<string>
                     : System.Diagnostics.SourceLevels.Information,
                 includeSeverityLevel: verbose);
         }
+
+        private static async Task<HashType> GetClientHashTypeAsync(
+            ArtifactHttpClientFactory factory,
+            VssConnection connection,
+            BlobStore.WebApi.Contracts.Client client,
+            IAppTraceSource tracer,
+            AgentTaskPluginExecutionContext context,
+            CancellationToken cancellationToken)
+        {
+            HashType hashType = ChunkerHelper.DefaultChunkHashType;
+
+            if (AgentKnobs.AgentEnablePipelineArtifactLargeChunkSize.GetValue(context).AsBoolean())
+            {
+                try
+                {
+                    var blobUri = connection.GetClient<ClientSettingsHttpClient>().BaseAddress;
+                    var clientSettingsHttpClient = factory.CreateVssHttpClient<IClientSettingsHttpClient, ClientSettingsHttpClient>(blobUri);
+                    ClientSettingsInfo clientSettings = await clientSettingsHttpClient.GetSettingsAsync(client, cancellationToken);
+                    if (clientSettings != null && clientSettings.Properties.ContainsKey(ClientSettingsConstants.ChunkSize))
+                    {
+                        HashTypeExtensions.Deserialize(clientSettings.Properties[ClientSettingsConstants.ChunkSize], out hashType);
+                    }
+                }
+                catch (Exception exception)
```

Comment:
>Exception [](http://example.com/codeflow?start=23&length=9)

what exception is expected here?  e.g. if none is expected, better not to handle it.  If a more specific exception is expected, narrow handling is better.  best would be to prevent the exception

catching the exception can make debugging difficult (though good we're logging the callstack)

Reply:
I totally agree that we should catch a more generic exception. I don't want the execution to fail if there is an exception (due to network issue, AFD, etc), hence to begin with I am just tracing it. The idea is to remove it eventually.